### DISCUSSION TITLE
Wezterm rendering bug?

### DISCUSSION THREAD

**Discussion by Lilja** [wezterm/wezterm#2476] [Q&A]
I have to copies of two different terminals, one being wezterm. The other being the windows terminal. Both use the same font(FiraCode NF), but one screws up the vim statusline bar. In neovim, I'm running `:echo b:editorconfig` which is an empty array in terminal, and instead `]        g` in wezterm. You can also see an opening bracket in the far right section of the neovim statusline bar.

![img](https://i.imgur.com/biFmwmE.png)

How can I investigate this further? I'm on windows.

**Comment by wez** [comment]
Please capture a terminal recording:

* Launch wezterm.  If possible, please use the default terminal size of 80x24 cells as it helps to keep things smaller and easier to manage.
* Inside that terminal run `wezterm record` to start a recording session.
* Run through your reproduction steps
* Then type `exit`
* You should see a message like:

```
*** Finished recording to /var/tmp/wezterm-recording-sF6B3u.cast.txt
```

* Attach the file that it produced to this issue.

The file is an asciicast (compatible with https://asciinema.org/) and can also be replayed using `wezterm replay`.

The terminal recording allows me to replicate what is being sent to the terminal without requiring me to install the same applications as you and replicate your configuration for everything.

**Reply by Lilja** [reply]
Is windows supported with wezterm record? Doesn't look like it: https://github.com/wez/wezterm/blob/main/wt-record#L9-L17

**Reply by wez** [reply]
`wezterm record` is part of `wezterm.exe` and works on all platforms.

**Reply by Lilja** [reply]
Thanks for clarifying that up! [Here's my cast](https://gist.github.com/Lilja/f20ecf5261bbc408af133c375ceab8aa)

**Reply by wez** [reply]
I got kinda swamped so am only just getting back to this.  From your recording, it looks like:

```
{'indent_style': 'space', 'indent_size': '2'}
```

is being printed into your status bar.

I don't know why; there's not enough context for me to understand what is really being attempted here.
My vim doesn't know anything about `b:editorconfig`, so I'm assuming that there's some vim plugin you're using that is doing something.  I don't know what is supposed to be happening and I don't really have time to dig in further, sorry!

**Reply by Lilja** [reply]
Thanks for taking some of your precious time! To give more context, I'm using [gpanders/editorconfig.nvim](https://github.com/gpanders/editorconfig.nvim) to lookup an `.editorconfig` file, which translates an editorconfig to vim's `swiftwidth` and `expandtab/noexpandtab` settings. The plugin will save contents to a buffer variable. I'm just using that as an example, since the first `{` of that echo will place it all the way to the right, as opposed to where the vim status bar is.

Here's me doing `:echo "lol"` inside vim:
![pic](https://i.imgur.com/0JAzhrR.png)

See how after the column/line number is 1:1, I have a doublequote, then the `lol`, a bunch of whitespace(?) and then another doublequote.

Here's another screencast where I just `echo "hi"` into vim after finding a file through `Telescope`: 
[wezterm-recording-rediE2.cast.txt](https://github.com/wez/wezterm/files/10395416/wezterm-recording-rediE2.cast.txt)
